### ISSUE TITLE
Neo4j 2.2 spatial withindistance where issue

### ISSUE THREAD

**Issue by Jempan** [neo4j/neo4j#4316]
Just started playing around with Neo4j 2.2. Awesome work! Unfortunately I ran in some problem using spatial-0.14-neo4j-2.2.0-M02 :( In my application I have a query which fetches the nearest users:

START n=node:geom('withinDistance:[42.0,1.0, 1000.0]') WHERE n:User RETURN n;

After updating to the latest version above query does not seem to care about label anymore and I get all nodes which has lat/lon and are added to the spatial layer back.

Thanks to Stefan at SO I got help with a workaround:
http://stackoverflow.com/questions/29319366/neo4j-2-2-spatial-withindistance-label-issue

**Comment by jexp** [comment]
Sorry for the delay.

Could you provide the query plan? By prefixing your query with `PROFILE` in the browser?

As a workaround:
1. try to prefix your query with `PLANNER RULE`
2. try to use

```
START n=node:geom('withinDistance:[42.0,1.0, 1000.0]') 
WITH n
WHERE n:User 
RETURN n;
```

**Comment by jexp** [comment]
This should be fixed in 2.2.1, see

https://github.com/neo4j/neo4j/pull/4346
